src/main/scala/piconot/parser/Parser.scala

```diff
@@ -0,0 +1,81 @@
+package piconot.parser
+
+import scala.util.parsing.combinator._
+import picolib.semantics._
+/**
+ * -----------
+ * Grammar
+ * -----------
+ * 
+ *          r Rule ::= st sr* -> c (st)
+ *          st State ::= Int
+ *          c Command ::= N | E | S | W | X
+ *          sr Surroundings ::== No (sr) | Nx (sr) | So (sr) ...
+ *  
+ */
+object PicoParser extends JavaTokenParsers {
+  
+
+    // parsing interface
+    def apply(s: String): ParseResult[List[Rule]] = parseAll(rule.*, s)
+
+    // rules
+    def rule: Parser[Rule] = startState~surroundings~"->"~command~(endState.?)~";" ^^ {
+      case startState~surroundings~"->"~command~endState~";" ⇒  Rule( 
+                                                          startState, 
+                                                          and(surroundings, blank), 
+                                                          command, 
+                                                          endState getOrElse startState
+                                                        )
+    }
+    def and(surroundingsList: List[Surroundings],toReturn: Surroundings): Surroundings = (surroundingsList,toReturn) match{
+      case (Nil,toReturn) => toReturn
+      case (No::rest,Surroundings(_,e,w,s)) => and(rest,Surroundings(Open,e,w,s))
+      case (Nx::rest,Surroundings(_,e,w,s)) => and(rest,Surroundings(Blocked,e,w,s))
+      case (So::rest,Surroundings(n,e,w,_)) => and(rest,Surroundings(n,e,w,Open))
+      case (Sx::rest,Surroundings(n,e,w,_)) => and(rest,Surroundings(n,e,w,Blocked))
+      case (Eo::rest,Surroundings(n,_,w,s)) => and(rest,Surroundings(n,Open,w,s))
+      case (Ex::rest,Surroundings(n,_,w,s)) => and(rest,Surroundings(n,Blocked,w,s))
+      case (Wo::rest,Surroundings(n,e,_,s)) => and(rest,Surroundings(n,e,Open,s))
+      case (Wx::rest,Surroundings(n,e,_,s)) => and(rest,Surroundings(n,e,Blocked,s))
+      case _ => throw new IllegalArgumentException("Bad argument, man")   
```

Comment:
Sure there is. :) It's caught by the default case in which an exception is thrown. Also I don't believe there's a valid way for users to represent `blank` (i.e., `Surroundings(Anything, Anything, Anything, Anything)` or the Picobot equivalent `****`) in our syntax&mdash;which is fine by me as I don't think there's any reason to need to represent it.